### PR TITLE
CI: virtual environnement to run integration tests using Cumulus Linux and libvirt

### PR DESCRIPTION
**Pull request by nqb**
# Description
- Add Cumulus Vagrantfile to manage a Cumulus Switch and a supplicant in a virtual environment with :
  - registration and isolation VLAN
  - RADIUS enforcement
  - Dynamic enforcement
  - inline enforcement
- Decrease memory used by VM in pipeline (httpd.admin now disabled)
- Refactor and improve `vagrant-wrapper.sh` to run tests

# Impacts 
Test stage in CI
FIXME

# Delete branch after merge
YES

# Checklist
- [x] Document the feature

# NEWS file entries
## New Features
* CI: virtual environnement to run integration tests using Cumulus Linux and libvirt

### CODE REVIEW

File: addons/vagrant/Vagrantfile [hunk 1]
```diff
@@ -40,18 +46,40 @@ Vagrant.configure("2") do |config|
...
 
         # registration
         # libvirt__forward_mode: "none" mean:
         # Allow traffic between guests. Deny inbound. Deny outbound.
-        srv.vm.network "private_network", ip: details['reg_ip'], netmask: details['reg_netmask'], libvirt__dhcp_enabled: false, libvirt__forward_mode: "none"
+        srv.vm.network "private_network",
+                       :libvirt__network_name => networks[1]['name'],
+                       :ip => networks[1]['subnet'],
+                       :libvirt__dhcp_enabled => false,
+                       :libvirt__forward_mode => 'none',
```
Comment (fdurand) [inline addons/vagrant/Vagrantfile]: Maybe nat if you want to test the passthrough

File: addons/vagrant/Vagrantfile [hunk 2]
```diff
@@ -40,18 +46,40 @@ Vagrant.configure("2") do |config|
...
+                       auto_config: false
 
         # isolation
-        srv.vm.network "private_network", ip: details['iso_ip'], netmask: details['iso_netmask'], libvirt__dhcp_enabled: false, libvirt__forward_mode: "none"
+        srv.vm.network "private_network",
+                       :libvirt__network_name => networks[2]['name'],
+                       :ip => networks[2]['subnet'],
+                       :libvirt__dhcp_enabled => false,
+                       :libvirt__forward_mode => 'none',
```
Comment (fdurand) [inline addons/vagrant/Vagrantfile]: Maybe nat if you want to test the passthrough

**Review by fdurand — APPROVED** [review]
Nothing wrong i can see. Only maybe enable nat on the reg/isol interfaces.

File: addons/vagrant/Vagrantfile (hunk 1 above)
Comment (nqb) [reply addons/vagrant/Vagrantfile]: Good catch, done.

File: addons/vagrant/Vagrantfile (hunk 2 above)
Comment (nqb) [reply addons/vagrant/Vagrantfile]: Good catch, done.

**Comment by nqb** [comment]
Docs updated, vagrant wrapper refactored and issues fixed in pipelines.

Can be review and merge if all is fine.

A new pipeline is running here: https://gitlab.com/inverse-inc/packetfence/-/jobs/483588188 with only INTEGRATION_TESTS part.

File: docs/integration_tests/README.asciidoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
+=== Integration tests
+
+The aim of integration tests is to test common PacketFence scenarios and
+integrations in a pipeline.
+
+Integration tests rely on:
+
+* <<_virtual_environment>>
+* a network topology built using `libvirt` footnote:[based on Cumulus Networks's project link:https://github.com/CumulusNetworks/cldemo-vagrant[cldemo]]
+* virtual machines different than PF to test integrations
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: PF -> PacketFence, everywhere

File: docs/integration_tests/README.asciidoc [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+==== Network Topology
+
+.Integration tests topology
+image::integration_tests/integration_tests_topology.png[scaledwidth="100%",alt="Integration tests topology"]
+
+This topology require a working `libvirt` setup (Virtualbox is not supported) and is defined as
+Infrastructure-as-code in two
+link:https://github.com/inverse-inc/packetfence/tree/devel/addons/vagrant/[Vagrantfiles].
+
+===== Initial provisioning
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: Provisioning

File: docs/integration_tests/README.asciidoc [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+`switch01` and `node01` are provisioned using Vagrant like this:
+
+* a first boot with basic shell provisioning to configure interfaces remap and networks
+* a second boot that handles interface remap
+* an Ansible provisioning
+
+Provisioning is deliberately minimalist to make the most during integration tests.
+
+===== Inventory informations
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: Information

File: docs/integration_tests/README.asciidoc [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+| Isolation | 172.17.3.0/24 | 3 
+| Inline | 172.17.6.0/24 | 6 
+| Internet | 192.168.121.0/24 | 100
+
+|===
+
+Management network is used to provision each virtual machine using Ansible to
+put them in a desired state.
+
+==== Virtual machines
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc:99]: Machines

File: docs/integration_tests/README.asciidoc [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+===== Active directory server
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: Directory Server

File: docs/integration_tests/README.asciidoc [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+Virtual machine called `ad`  will be auto-provision with:
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: auto-provisioned

File: docs/integration_tests/README.asciidoc [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+* a DNS domain
+* an Active Directory domain
+* AD CS services with auto-enrollment using a GPO
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: CS? Don't use the acronym if it's the first time you use it

File: docs/integration_tests/README.asciidoc [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+* a packetfence account (Domain users)
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: if the account name is "packetfence" say it clearly

File: docs/integration_tests/README.asciidoc [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+===== Switch01
+
+`switch01` is a virtual switch that uses link:https://docs.cumulusnetworks.com/version/cumulus-linux-37/[Cumulus Linux 3.7] system.
+
+This switch is configured using a
+link:https://docs.cumulusnetworks.com/cumulus-linux-37/Layer-2/Ethernet-Bridging-VLANs/[VLAN
+aware-bridge] to manage VLAN on interfaces (recommended approach)
+
+`hostapd` is the daemon responsible to sent 802.1X and MAC Authentication
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: to send

File: docs/integration_tests/README.asciidoc [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+link:https://docs.cumulusnetworks.com/version/cumulus-linux-37/Layer-1-and-Switch-Ports/802.1X-Interfaces/[802.1X
+Cumulus Linux documentation]). We configured this daemon using Ansible.
+
+===== Node01
+
+`node01` is a Debian Buster virtual machine used as node to test PacketFence
+features like 802.1X, MAC authentification, registration, isolation and inline
+scenarios.
+
+This machine has an Internet access during it's first boot. After the reboot, this machine is only
```
Comment (extrafu) [inline docs/integration_tests/README.asciidoc]: its

**Comment by nqb** [comment]
Typos fixed @extrafu, need to fix issue with `ad` machine on CI server before merge. Working on it.

**Comment by nqb** [comment]
Issue fixed on CI server, running a new pipeline to test if it works as it should.

**Comment by nqb** [comment]
@extrafu & @fdurand, this PR can be merged. If there is other issues detected, I will fix them post-merge.

**Pull request merged**